src/receive_imf.rs

```diff
@@ -105,10 +101,33 @@ pub(crate) async fn receive_imf_inner(
 
     let mut mime_parser =
         match MimeMessage::from_bytes_with_partial(context, imf_raw, is_partial_download).await {
-            Err(err) => {
+            Err(ParserError::Malformed(err)) => {
                 warn!(context, "receive_imf: can't parse MIME: {}", err);
-                return Ok(None);
+
+                let msg_ids;
+                if !rfc724_mid.starts_with(GENERATED_PREFIX) {
+                    let row_id = context
+                        .sql
+                        .execute(
+                            "INSERT INTO msgs(rfc724_mid, chat_id) VALUES (?,?)",
+                            paramsv![rfc724_mid, DC_CHAT_ID_TRASH],
+                        )
+                        .await?;
+                    msg_ids = vec![MsgId::new(u32::try_from(row_id)?)];
+                } else {
+                    return Ok(None);
+                    // We don't have an rfc724_mid, there's point in adding a trash entry
+                }
+
+                return Ok(Some(ReceivedMsg {
+                    chat_id: DC_CHAT_ID_TRASH,
+                    state: MessageState::Undefined,
+                    sort_timestamp: 0,
+                    msg_ids,
+                    needs_delete_job: false,
+                }));
             }
+            Err(ParserError::Sql(err)) => return Err(err),
```

Comment:
This is the most important part of e2d706ee: If the email was malformed, create a trash entry. If there was a Sql error, return the error.